Sources/Amplitude/AMPUtils.m

```diff
@@ -96,6 +96,13 @@ + (id)makeJSONSerializable:(id)obj {
     return str;
 }
 
++ (NSMutableDictionary *)removeNilValues:(NSDictionary *)dict {
+    NSMutableDictionary *d = [dict mutableCopy];
+    NSArray *keysForNullValues = [d allKeysForObject:[NSNull null]];
+    [d removeObjectsForKeys:keysForNullValues];
+    return d;
+}
+
```

Comment:
I think it'd be be more performant to just ignore a key if it's null when iterating over them, to prevent having to loop over all keys twice. Let me know if that is unfeasible in this language/the way the code is set up, because I'm unfamiliar with the sdk.

Reply:
Sorry I think I wasn't clear. I believe `addEntriesFromDictionary` would iterate over the dict once. The new `removeNilValues` also iterates over all the keys.

I was thinking in `addEntriesFromDictionary`, whenever we add the value to the dictionary, we would just do something like 
```
if (value != null) {
   add to dict
}
```
and remove `removeNilValues` entirely. Let me know if I'm understanding `addEntriesFromDictionary` wrong, because I'm not sure of the actual body of that function